apps/emqx_resource/src/emqx_resource_worker.erl

```diff
@@ -680,9 +696,9 @@ reply_caller_defer_metrics(Id, ?REPLY(From, _, HasBeenSent, Result), QueryOpts)
     {ShouldAck, PostFn}.
 
 handle_query_result(Id, Result, HasBeenSent) ->
-    {ShouldBlock, PostFn} = handle_query_result_pure(Id, Result, HasBeenSent),
+    {_ShouldBlock, PostFn} = handle_query_result_pure(Id, Result, HasBeenSent),
     PostFn(),
-    ShouldBlock.
+    ok.
```

Comment:
I think it can be confusing to ignore the result above?

Reply:
it's not used anywhere from this function's return.

Reply:
But it's different from the behavior of the pure version, which could be misleading when touching this module/fn in the future.  If the result is not wanted at the call site now, maybe we can just ignore it there?